AirportTicketBookingSystem/AirportTicketBookingSystem/Repositories/FlightRepository.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            using var writer = new StreamWriter(_filePath);
+            using var csv = new CsvWriter(writer, CultureInfo.InvariantCulture);
+            csv.WriteRecords(flights);
+        }
+
+        public List<Flight> SearchAvailableFlights(string departureCountry = null, string destinationCountry = null,
+            string departureAirport = null, string arrivalAirport = null, DateTime? departureDateFrom = null,
+            DateTime? departureDateTo = null, TravelClass? seatClass = null, decimal? minPrice = null, decimal? maxPrice = null)
+        {
+            var allFlights = GetAllFlights();
+            var allBookings = _bookingRepository.GetAllBookings();
+
+            FlightBookingUtilities.AttachBookings(allFlights, allBookings);
+
+            var query =
+            from f in allFlights
+            where (string.IsNullOrEmpty(departureCountry) || f.DepartureCountry.Equals(departureCountry, StringComparison.OrdinalIgnoreCase))
+            where (string.IsNullOrEmpty(destinationCountry) || f.DestinationCountry.Equals(destinationCountry, StringComparison.OrdinalIgnoreCase))
+            where (string.IsNullOrEmpty(departureAirport) || f.DepartureAirport.Equals(departureAirport, StringComparison.OrdinalIgnoreCase))
+            where (string.IsNullOrEmpty(arrivalAirport) || f.ArrivalAirport.Equals(arrivalAirport, StringComparison.OrdinalIgnoreCase))
+            where (!departureDateFrom.HasValue || f.DepartureDate >= departureDateFrom.Value)
+            where (!departureDateTo.HasValue || f.DepartureDate <= departureDateTo.Value)
+            where (!seatClass.HasValue || f.Bookings.Any(b => b.SeatClass.Name == seatClass.Value))
+            where (!minPrice.HasValue || f.Bookings.Any(b => b.SeatClass.CalculatePrice() >= minPrice.Value))
+            where (!maxPrice.HasValue || f.Bookings.Any(b => b.SeatClass.CalculatePrice() <= maxPrice.Value))
+            select f;
+
+            return query.ToList();
+        }
+
+        public List<string> ImportFlightsFromCsv(string csvFilePath)
+        {
+            var importedFlights = GetAllFlights(csvFilePath);
+
+            if (!importedFlights.Any())
+                throw new ArgumentException("No flights found in the CSV file");
+
+            var existingFlights = GetAllFlights();
+            var errors = new List<string>();
+
+            foreach (var flight in importedFlights)
+            {
+                var flightErrors = ValidateFlightForImport(flight, existingFlights);
+
+                if (flightErrors.Any())
+                {
+                    errors.AddRange(flightErrors);
+                    continue; 
+                }
+                flight.Id = IdGenerator.GenerateNewId(existingFlights);
+                existingFlights.Add(flight);
+            }
+            Save(existingFlights);
+
+            return errors;
+        }
+
+        public bool IsDuplicateFlight(Flight flight, List<Flight> existingFlights)
```

Comment:
same thing, this method is not using the db storage at all, can go to the service layer